src/main/java/htsjdk/variant/vcf/VCFFileReader.java

```diff
@@ -16,105 +16,266 @@
 import java.io.Closeable;
 import java.io.File;
 import java.io.IOException;
+import java.nio.file.Path;
 
 /**
  * Simplified interface for reading from VCF/BCF files.
+ *
  */
+
 public class VCFFileReader implements Closeable, Iterable<VariantContext> {
 
-	private final FeatureReader<VariantContext> reader;
-
-	/**
-	 * Returns true if the given file appears to be a BCF file.
-	 */
-	public static boolean isBCF(final File file) {
-		return file.getAbsolutePath().endsWith(".bcf");
-	}
-
-	/**
-	 * Returns the SAMSequenceDictionary from the provided VCF file.
-	 */
-	public static SAMSequenceDictionary getSequenceDictionary(final File file) {
-	    try (final VCFFileReader vcfFileReader = new VCFFileReader(file, false)) {
-		    return vcfFileReader.getFileHeader().getSequenceDictionary();
-		}
-	}
-
-    /** Constructs a VCFFileReader that requires the index to be present. */
-	public VCFFileReader(final File file) {
-		this(file, true);
-	}
-
-    /** Constructs a VCFFileReader with a specified index. */
+    private final FeatureReader<VariantContext> reader;
+
+    /**
+     * Returns true if the given file appears to be a BCF file.
+     *
+     * @deprecated in favor of {@link VCFFileReader#isBCF(Path)}
+     */
+    @Deprecated
```

Comment:
We've typically been avoiding deprecating the file methods so far, but it would be nice to move away from them.  Probably fine to do so here.

Reply:
That said, if we deprecate them, we should remove uses of them except for direct tests.  Lots of test code is using the `File` constructors.

Reply:
That seems like a larger effort than this PR is meant to do...reversing the deprecations...

Reply:
Perhaps this should be on the roadmap for htsjdk3?

Reply:
Is there any plan appart of #520 to implement htsjdk3?